src/frontend/src/handler/query.rs

```diff
@@ -90,16 +91,24 @@ fn must_run_in_local_mode(bound: &BoundStatement) -> bool {
     must_local
 }
 
+pub struct BatchQueryPlanResult {
+    pub(crate) plan: PlanRef,
+    pub(crate) query_mode: QueryMode,
+    pub(crate) schema: Schema,
+    pub(crate) dependent_views: Vec<ViewId>,
```

Comment:
It looks a little bit confusing why we treat `dependent_views` differently here. 🥹

Is it possible to resolve all relations here? (Ok to add a TODO) If not or hard, maybe add some comments to explain the reason.

Reply:
Yes exactly. I'm thinking of putting all the resolving of dependency information together on the frontend. Part of it is currently being done in fragmenter. I will add a TODO here and fix it in another PR.

Reply:
I unified resolving dependent relations in the frontend while binding `table`/`view`/`source`. PTAL. Cc @xx01cyx